### PR TITLE
feat(cli): create workspace using parameters from existing workspace

### LINKED ISSUE #9923: Feature: Duplicate workspace
**Extracted from user conversation**

I find myself wanting to launch new workspaces where the configuration (i.e. input params) is nearly identical to a workspace that already exists.

One scenario: workspace in us-east-1, but there are aws issues, so now I want to relaunch a similar workspace in us-west-2 with all the same inputs (i.e. repo configuration, instance size, etc.).

I have to manually copy all the workspace settings to a new workspace, which is a tad tedious.

**Possible solutions**

In the UI we could have:
- A button called “Duplicate” or “Clone” on the workspace page
- When clicking on the button, it will display a dialog with a form to override, if the user wants, the workspace settings
- Confirming, will create a new workspace with the new/overridden settings

In the CLI:
We could do something similar, `coder duplicate  [some-user.]some-workspace` and it would operate like `coder create -t template` but with all the defaults populated from the source workspace

### PR DESCRIPTION
**Pull request by mtojek**
Fixes: https://github.com/coder/coder/issues/9923

This PR adds support for duplicating workspace using CLI. It is implemented as part of the `coder create` command according to this [design](https://github.com/coder/coder/issues/9923#issuecomment-1759551552) (thanks @Parkreiner!).

```bash
./scripts/coder-dev.sh create --copy-parameters-from=workspace-5 workspace-6-copy
```

### CODE REVIEW

File: cli/create_test.go [hunk 1]
```diff
@@ -416,6 +416,60 @@ func TestCreateWithRichParameters(t *testing.T) {
...
+		client := coderdtest.New(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
+		owner := coderdtest.CreateFirstUser(t, client)
+		member, _ := coderdtest.CreateAnotherUser(t, client, owner.OrganizationID)
+		version := coderdtest.CreateTemplateVersion(t, client, owner.OrganizationID, echoResponses)
+		coderdtest.AwaitTemplateVersionJobCompleted(t, client, version.ID)
+
+		template := coderdtest.CreateTemplate(t, client, owner.OrganizationID, version.ID)
+
+		// Firstly, create a regular workspace using template with parameters.
+		inv, root := clitest.New(t, "create", "my-workspace", "--template", template.Name, "-y",
```
Comment (johnstcn) [inline cli/create_test.go:432]: NB: we now have [dbfake.WorkspaceWithAgent](https://github.com/coder/coder/blob/main/coderd/database/dbfake/dbfake.go#L42) if you want to speed things up

File: cli/create_test.go [hunk 2]
```diff
@@ -416,6 +416,60 @@ func TestCreateWithRichParameters(t *testing.T) {
...
 		assert.ErrorContains(t, err, "Did you mean: "+firstParameterName)
 	})
+
+	t.Run("CopyParameters", func(t *testing.T) {
+		t.Parallel()
+
+		client := coderdtest.New(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
+		owner := coderdtest.CreateFirstUser(t, client)
+		member, _ := coderdtest.CreateAnotherUser(t, client, owner.OrganizationID)
+		version := coderdtest.CreateTemplateVersion(t, client, owner.OrganizationID, echoResponses)
```
Comment (johnstcn) [inline cli/create_test.go:426]: I'd like to see another test where the workspace is on an older template version, and the "cloned" workspace is on the exact same version.

File: cli/testdata/coder_create_--help.golden [hunk 3]
```diff
@@ -14,6 +14,9 @@ OPTIONS:
           Specify automatic updates setting for the workspace (accepts 'always'
           or 'never').
 
+      --copy-parameters string, $CODER_WORKSPACE_COPY_PARAMETERS
```
Comment (johnstcn) [inline cli/testdata/coder_create_--help.golden:17]: Nit: `--copy-parameters-from`? Up to you.

**Review by johnstcn — APPROVED** [review]
Looks good to me! Just an edge case I'd like to see tested, and some potential speed-ups in creation.

File: cli/create_test.go (hunk 1 above)
Comment (mtojek) [reply cli/create_test.go:432]: `WorkspaceWithAgent` looks efficient, but I need a workspace with build parameters and a corresponding template with rich parameters. Not sure if the function supports it.

Comment (johnstcn) [reply cli/create_test.go:432]: 😢 🎺

File: cli/testdata/coder_create_--help.golden (hunk 3 above)
Comment (mtojek) [reply cli/testdata/coder_create_--help.golden]: Done

File: cli/create_test.go (hunk 2 above)
Comment (mtojek) [reply cli/create_test.go:426]: A bit of surgery, and I think I got it.

**Review by johnstcn — APPROVED** [review]

**Comment by matifali** [comment]
May be we should call it `coder clone`. Thoughts?
Or `coder create --clone=old-workdoace new-workspace`

We can put a disclaimer that it just copies the parameters.

**Comment by mtojek** [comment]
@matifali There were concerns raised around "clone" word: [comment](https://github.com/coder/coder/issues/9923#issuecomment-1757805980)

**Pull request merged**